src/training/train.py

```diff
@@ -232,22 +255,24 @@ def evaluate(model, data, epoch, args, tb_writer=None):
             f.write(json.dumps(metrics))
             f.write("\n")
 
-    if args.wandb:
+    if args.wandb and is_master(args):
         assert wandb is not None, 'Please install wandb.'
         for name, val in metrics.items():
             wandb.log({f"val/{name}": val, 'epoch': epoch})
 
     return metrics
 
 
-def get_metrics(image_features, text_features, logit_scale):
+def get_metrics(image_features, text_features, logit_scale, args):
```

Comment:
nit pick but maybe better to pass a more specific argument e.g. `gather_tensor=args.distributed_evaluation`?

Reply:
> In the pytorch imagenet example for distributed imagenet eval they have an `aux_val_loader` to handle the case where the test set size is not divisible by num_gpus - do we need to have this? https://github.com/pytorch/examples/blob/main/imagenet/main.py#L396-L402

Thanks for the link, not sure why they do `drop_last=True` on val_loader (not used here), probably to avoid having a GPU worker with much fewer examples than the others? so rather, they seem to do drop_last, and compute the last few examples validation performance in all GPU workers.

Reply:
> nit pick but maybe better to pass a more specific argument e.g. `gather_tensor=args.distributed_evaluation`?

Thanks I agree, I think it's better